nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -158,9 +158,7 @@ private static String getDefaultProviderKey() {
     }
 
     private void initializeSensitivePropertyProviderFactory() {
-        if (sensitivePropertyProviderFactory == null) {
-            sensitivePropertyProviderFactory = new AESSensitivePropertyProviderFactory(keyHex);
-        }
+        sensitivePropertyProviderFactory = new AESSensitivePropertyProviderFactory(keyHex);
```

Comment:
@alopresto won't this still have race conditions if more than one thread does NiFiPropertiesLoader.withKey() at the same time?